vetomint/src/progress.rs

```diff
@@ -1,9 +1,158 @@
 use super::*;
 
-pub fn progress(
+/// TODO: we have to implement the following missing logics
+/// - on-proposal
+/// - on-4f-favor-prevote-propose-step
+/// - on-4f-favor-prevote-prevote-step
+/// - on-4f-nil-prevote
+/// - on-5f-precommit
+/// - on-4f-favor-precommit
+/// - OnTimeoutPrecommit
+pub(super) fn progress(
+    height_info: &HeightInfo,
+    state: &mut ConsensusState,
+    event: ConsensusEvent,
+) -> Option<Vec<ConsensusResponse>> {
+    let result = if state.waiting_for_proposal_creation {
+        if let ConsensusEvent::BlockProposalCreated {
+            proposal, round, ..
+        } = event
+        {
+            if state.round != round {
+                return None;
```

Comment:
Why it returns None?
if we got BlockProposalCreated event with wrong round,
it represents a) we got wrong proposal, or b) we were in wrong round.
in case of a), None would suit for it, but I wonder case b).
is b) solved by other mod?

Reply:
If the node is in a wrong round, the only way of getting out of it is just providing valid prevotes and precommits for the missing rounds. There is nothing to do in this case, anyway.